CONTRIBUTING.md

```diff
@@ -0,0 +1,169 @@
+are important to the project's success.
+
+
+## Getting started, building, and testing
+
+If you haven't already, take a look at the project's
+[README.md file](README.md)
+and try adding type annotations to your file and type-checking it with
+a tool like [Mypy](http://www.mypy-lang.org/),
+[pytype](https://github.com/google/pytype), or
+[PyCharm](https://www.jetbrains.com/pycharm/).
+
+
+## Discussion
+
+If you've run into behavior in the type checker that suggests the type
+stubs for a given library are incorrect or incomplete, or a library you
+depend on is missing type annotations, we want to hear from you!
+
+Our main forum for discussion is the project's [GitHub issue
+tracker](https://github.com/python/typeshed/issues).  This is the right
+place to start a discussion of any of the above or most any other
+topic concerning the project.
+
+For less formal discussion, try the Mypy chat room on
+[gitter.im](https://gitter.im/python/mypy).  Some Mypy core developers
+are almost always present; feel free to find us there and we're happy
+to chat.  Substantive technical discussion will be directed to the
+issue tracker.
+
+### Code of Conduct
+
+Everyone participating in the typeshed community, and in particular in
+our issue tracker, pull requests, and IRC channel, is expected to treat
+other people with respect and more generally to follow the guidelines
+articulated in the [Python Community Code of
+Conduct](https://www.python.org/psf/codeofconduct/).
+
+
+## Submitting Changes
+
+Even more excellent than a good bug report is a fix for a bug, or the
+implementation of a much-needed stub. We'd love to have
+your contributions.
+
+We use the usual GitHub pull-request flow, which may be familiar to
+you if you've contributed to other projects on GitHub.  For the
+mechanics, see [Mypy's git and GitHub workflow help page](https://github.com/python/mypy/wiki/Using-Git-And-GitHub),
+or [GitHub's own documentation](https://help.github.com/articles/using-pull-requests/).
+
+Anyone interested in type stubs may review your code.  One of the core
+developers will merge your pull request when they think it's ready.
+For every pull request, we aim to promptly either merge it or say why
+it's not yet ready; if you go a few days without a reply, please feel
+free to ping the thread by adding a new comment.
+
+At present the core developers are (alphabetically):
+* David Fisher (@ddfisher)
+* Łukasz Langa (@ambv)
+* Jukka Lehtosalo (@JukkaL)
+* Matthias Kramm (@matthiaskramm)
+* Greg Price (@gnprice)
+* Guido van Rossum (@gvanrossum)
+
+
+## Preparing Changes
+
+Before you begin: if your change will be a significant amount of work
+to write, we highly recommend starting by opening an issue laying out
+what you want to do.  That lets a conversation happen early in case
+other contributors disagree with what you'd like to do or have ideas
+that will help you do it.
+
+If you want to submit type stubs for a new library, you need to
```

Comment:
I'd like this section to have more prominence. I'd also like it to have a section heading we can deep-link to so as to remind people who submit such a stub without first acquiring permission. Also, can you deep-link to the section in PEP 484 where this is brought up? Even there, there's not that much explanation of *why* we require this -- perhaps we can add that explanation here. One of the reasons is to allay some package owners' worries that bugs in the stubs would be mistaken by users as problems with their package and cause spurious bugs to be opened against their package. Another reason is that some package owners prefer to distribute their own stubs, or include inline annotations in their code.